cdap-watchdog/src/main/java/co/cask/cdap/logging/context/WorkflowProgramLoggingContext.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.logging.context;
+
+import co.cask.cdap.proto.ProgramType;
+
+/**
+ * Logging context to be used by programs running inside Workflow.
+ */
+public class WorkflowProgramLoggingContext extends WorkflowLoggingContext {
+  /**
+   * Constructs logging context.
+   *
+   * @param namespaceId namespace id
+   * @param applicationId application id
+   * @param workflowId workflow id
+   * @param runId run id of the application
+   * @param programType type of the program. Currently only MapReduce and Spark is supported.
+   * @param programName name of the program
+   */
+  public WorkflowProgramLoggingContext(String namespaceId, String applicationId, String workflowId, String runId,
+                                       ProgramType programType, String programName) {
+    super(namespaceId, applicationId, workflowId, runId);
+    switch (programType) {
+      case MAPREDUCE:
+        setSystemTag(MapReduceLoggingContext.TAG_MAP_REDUCE_JOB_ID, programName);
```

Comment:
When I meant record - it was while publishing logs so that we can apply a runId filter while reading the logs if required later. Let's say a workflow runs a map reduce twice. We won't be able to figure out what log lines are for which run of a map reduce solely based on the names, right?